misc/benchmark_mode.rb

```diff
@@ -0,0 +1,367 @@
+# frozen_string_literal: true
+
+require "etc"
+require "open3"
+
+module BenchmarkMode
+  BENCH_SET = "bench"
+  OTHER_SET = "other"
+
+  module Helpers
+    extend self
+
+    # Transform a list like "0,2-4,7" into an array of integers.
+    def list_to_ints(list)
+      list
+        &.split(',')
+        &.flat_map { |s| a, b = s.split('-').map(&:to_i); b ? a.upto(b).to_a : [a] }
+    end
+  end
+
+  class << self
+    # Enable all available settings.
+    def engage!(nice: nil)
+      Nice.renice_process_group!(nice || -15, Process.getpgrp)
```

Comment:
Should comment on what this does

Reply:
There are some comments below where the module is defined but yeah, I can add something here, too.